src/fireedge/src/modules/utils/parser/vmTemplateFilter.js

```diff
@@ -0,0 +1,810 @@
+    get(correctionMap, section, []),
+    get(correctionMap, section, [])
+  )
+
+  if (type === 'NIC') {
+    sectionModifications.forEach((value, index) => {
+      const indexAlias = value.__aliasIndex__
+      if (indexAlias !== undefined)
+        wrappedExistingData[index] = { ...originalData.NIC_ALIAS[indexAlias] }
+
+      const indexPci = value.__aliasPci__
+      if (indexPci !== undefined)
+        wrappedExistingData[index] = { ...originalData.PCI[indexPci] }
+    })
+  } else if (type === 'NIC_ALIAS') {
+    sectionModifications.forEach((value, index) => {
+      const indexNic = value.__nicIndex__
+      if (indexNic !== undefined)
+        wrappedExistingData[index] = { ...originalData.NIC[indexNic] }
+      const indexPci = value.__aliasPci__
+      if (indexPci !== undefined)
+        wrappedExistingData[index] = { ...originalData.PCI[indexPci] }
+    })
+  } else if (type === 'PCI') {
+    sectionModifications.forEach((value, index) => {
+      const indexNic = value.__nicIndex__
+      if (indexNic !== undefined)
+        wrappedExistingData[index] = { ...originalData.NIC[indexNic] }
+      const indexAlias = value.__aliasIndex__
+      if (indexAlias !== undefined)
+        wrappedExistingData[index] = { ...originalData.NIC_ALIAS[indexAlias] }
+    })
+  }
+
+  // Iterate over the final data
+  const modifiedData = formData.extra[type].map((item, index) => {
+    // Check if the index of the item it's on the modifications map and has value
+    if (
+      index < sectionModifications.length &&
+      sectionModifications[index] !== null
+    ) {
+      let itemModifications = {}
+
+      // If the type is PCI and not has a TYPE attribute, the pci is from the inputOutput section. In other case, is from network section. So in the second case, we have to get the attribute from advanced step, and not in the first case.
+      if (type === 'PCI' && !item?.TYPE) {
+        // Get the fields where the modifications were done
+        itemModifications = {
+          ...sectionModifications[index],
+          ...alwaysInclude,
+        }
+      } else {
+        // Get the fields where the modifications were done
+        itemModifications = Object.keys(sectionModifications[index])?.reduce(
+          (acc, key) => ({
+            ...acc,
+            ...sectionModifications[index][key],
+            ...alwaysInclude,
+          }),
+          {}
+        )
+      }
+
+      // Delete keys on existing data that are marked to delete and not exists on formData or are marked to update and not exists on formData
+      if (wrappedExistingData && wrappedExistingData[index])
+        Object.entries(itemModifications)
+          .filter(
+            ([key, value]) =>
+              (value.__delete__ && !item[key]) || (value && !item[key])
+          )
+          .forEach(
+            ([key, value]) =>
+              wrappedExistingData[index][key] &&
+              delete wrappedExistingData[index][key]
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected that you are using `delete` to remove items from an array. If we use `delete`, it will delete the object property, but will not reindex the array or update its length. It leaves `undefined` holes in the array. Therefore, we recommend you to use `splice` to remove items from an array.

[Learn more](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete)